src/arvgvdevice.c

```diff
@@ -842,9 +842,9 @@ arv_gv_device_get_persistent_ip (ArvGvDevice *gv_device,
 /**
  * arv_gv_device_set_persistent_ip:
  * @gv_device: a #ArvGvDevice
- * @ip: IPv4 address
- * @mask: Netmask
- * @gateway: Gateway IPv4 address
```

Comment:
Please use the (optional) annotation here:
```
* @ip: (optional): IPv4 address
```

Reply:
Try: `@ip: (out) (optional): IPv4 address

Reply:
Ok, thanks but in the case of `arv_gv_device_set_persistent_ip` this is an input parameter.

Reply:
Ah. (nullable) should work then.

Reply:
Should I keep the `, %NULL to ignore` at the end of the line?

Reply:
Here is the annotation documentation: https://gi.readthedocs.io/en/latest/annotations/giannotations.html

And indeed (nullable) should be used for (in) parameters.

Reply:
Don't keep `%NULL% to ignore`, it is redundant with the introspection annotation. gi-docgen should translate the annotation into a full text explanation.